### ISSUE TITLE
[Feature Request] refill trailer from exotics fillType and/or exotics vehicle...

### ISSUE THREAD

**Issue by Anonymous-any** [Stephan-S/FS19_AutoDrive#1717]
**AutoDrive Version: 1.1.0.7
**Release or Custom Zip:** Release  
**Map (If applicable): Yagodnoe 2.5.1
**Vehicle / Type (If applicable): trailer, bus
** Singleplayer  

Hello @AD team,

Could you have a fix for this method ?
Here I have to press "start refill" of course, then the menu showed then I have to press enter again, cause the menu showed for select the fruit but it cannot choose any other from it, as the trigger is what it is... maybe the map maker doesn't do a good job ? (I mean  showed a menu only for 1 fillType is stupid...) and/or maybe because the vehicle only can do this fillType. But it's like that.
Also in this map there lots exotics fruits I must say, and another special is "work" for worker, I have a bus for it but same explain as above it can only do one fillType and it pass on the refill trigger without fill and do his route.

Regard
![1](https://user-images.githubusercontent.com/25174503/101921038-d18d5600-3bcc-11eb-93b7-9848e3bfd3f5.png)
![2](https://user-images.githubusercontent.com/25174503/101921143-f08be800-3bcc-11eb-8e8c-382f5e528045.png)

**Comment by Axel32019** [comment]
@Anonymous-any Hello Ano, Please provide links to the fill station, trailer, or map if fill station is build-in the map.

**Comment by Anonymous-any** [comment]
Hello Axl,

Well yes it's build-in map, but it will go a bit complicated because I had to do some few modification for this trailer work.
The reason is because the one vehicle provided with the map don't take AD and CP usage. So I by-pass with adapt this trailer with my low knowledge. (Finally adapt for fillType and trigger).
So the trailer link : https://fs19.net/farming-simulator-2019-mods/trucks/concrete-mixer-v0-1/
https://fs19.net/download-mod/59210
but I can upload my own amended somewhere idk...

For the map a bit same story as above, but I don't touch any trigger, only lots translations, missions, bad spawns or a some tired thing like splash screens at every loads...and other inconsistency wrong path for vehicle texture I dislike, 2 or 3 bullshit, cams...nothing too bad. I can too add you my few translated files to my repo idk...ask me
Map link : https://www.modhoster.com/mods/village-yagodnoe-ussr
https://www.modhoster.com/file/C9LId

Another set here for the special bus "work", diesel trailer tank, and other things : https://sharemods.com/d5qasrnxi4jt/Yagodnoe_mods.7z.html but not linked to the page tittle maybe :)

Some vids that helped me to try to understand some things (and it's not over ^^).
https://www.youtube.com/watch?v=taEqD8DBvJ0
https://www.youtube.com/watch?v=Jh5G4rTxBrI
https://www.youtube.com/watch?v=SzCjDN0IRos
they seem are rare and fortunately they are there.

Br

**Comment by Axel32019** [comment]
Hello Ano,
Up to this point I could not remember an unload construction like in this map. 
The concrete is available as pallet and as loose material if I understand it correct?
What happen if you change the GBI.xml to only have the loose material unloading?
```
<outputMethods>
<loadingTrigger name="LoadingTriggerC" />
</outputMethods>
```
The trailer has filltype category bulk, so in the maps_fillTypes.xml bulk should include concrete to match, right?
`<fillTypeCategory name="BULK" >CONCRETE ....`
With this you selected a big project to get it to work!

**Comment by Anonymous-any** [comment]
Hello @Axel32019 ,
Yes someone "Stronger" (not the origin mapmaker I think it was Aleksey Potanin) been added lots Factory's and change.

In fact I got caught up in the game because I was used the 1.4 old version of this map (if I good remember), then I updated it and lost all my old save (not AD path). And then I redo a new game with this one when I been understand the change that will not be compatible with the old...
But there are lots issues like the bridge (good idea) but as example, the trigger to refill materials disappear when you start construction ^^. So to refill when it stop I have to manually add materials in the file directly... I think no lots peoples don't bother with that :p.
And because I play real time I'm not sure the other constructible factory will work well for this point.
Others bugs I see are to some tank trailer where the fillType "MASUT, BIOGAS, and others water trigger factory, bref..." simply not exist as the log say (I think it's solvable into the maps_fillTypes.xml as well) so Idk if I will keep this map but I try because I think it's not bad and I always try to imagine what the mapper wanted to do to fix and gave it more compatibility with my mods but it's not easy. Back to you, Idk why you say me "selected a big project to get it to work!" ?
> 
> 
> The trailer has filltype category bulk, so in the maps_fillTypes.xml bulk should include concrete to match, right?
> `<fillTypeCategory name="BULK" >CONCRETE ....`
![1](https://user-images.githubusercontent.com/25174503/102650179-c2794b80-416a-11eb-855e-fa2ee775affd.png)

Good new CONCRETE bulk work now ^^, you are really good, I think I will leave it as you do it and you method will serve me certainly me again in the future! 
For the concrete pallet I saw it in shop but I did not know it can produce (stay now where to sell it ...). I don't spawn it at time with the factory I will test.
```
<outputMethods>
<loadingTrigger name="LoadingTriggerC" />
</outputMethods>
```
I now don't really understand why to do that bulk **only** what the link with pallets and the bulk fillType ?
So, do you know what I can do now for AD and "work" fillType with the bus managed by AD ?
I think finally maybe change the tittle to BUG feature again ?

Br :)

**Comment by Axel32019** [comment]
@Anonymous-any Hello Ano,
Was busy with other topics...
What you mean with your statements - sometimes it is hard for me to understand.
```
I now don't really understand why to do that bulk only what the link with pallets and the bulk fillType ?
So, do you know what I can do now for AD and "work" fillType with the bus managed by AD ?
I think finally maybe change the tittle to BUG to -> feature request again ?
```

**Comment by Anonymous-any** [comment]
Hello @Axel32019 , happy new year 21 in advance !

Well it's me np ! after you say me that, I tried to understanding what is the link with "concrete pallets" you make me removed in file right like your post 13 day ago :
What happen if you change the GBI.xml to only have the loose material unloading?

<outputMethods>
<loadingTrigger name="LoadingTriggerC" />
</outputMethods>

So that why my brain search to it but it's ok I set it back and concrete pallet can spawn np.
For the rest about "work" fillType, I thinking it's was linked to this request too meaning if AD can load-unload tons of exotics/customs fillType automatically from the mapmakers it's good explained now ? cause finally AD can do automatic with bulk type right, but it seem not work with custom fillType menu screen dunno (tried too with autoloader pallets truck-trailers) but AD need some help to validate the load. I know I already saw tickets about AD pallets requests ;).
 Br
